watson/cli.py

```diff
@@ -476,23 +483,29 @@ def report(watson, current, from_, to, projects,
                 longest_tag = max(len(tag) for tag in tags or [''])
 
                 for tag in tags:
-                    click.echo('\t[{tag} {time}]'.format(
+                    lines.append('\t[{tag} {time}]'.format(
                         time=style('time', '{:>11}'.format(format_timedelta(
                             datetime.timedelta(seconds=tag['time'])
                         ))),
                         tag=style('tag', '{:<{}}'.format(
                             tag['name'], longest_tag
                         )),
                     ))
-            click.echo()
+            lines.append("")
 
         if len(projects) > 1:
-            click.echo('Total: {}'.format(
+            lines.append('Total: {}'.format(
                 style('time', '{}'.format(format_timedelta(
                     datetime.timedelta(seconds=report['time'])
                 )))
             ))
 
+    if pager or (pager is None and
+                 watson.config.getboolean('options', 'pager')):
+        click.echo_via_pager('\n'.join(lines))
+    else:
+        click.echo('\n'.join(lines))
+
```

Comment:
I think we should avoid buffering all the lines when in paging mode. The way to do it is not obvious, but maybe something like that would work:
```
lines = []

if pager or (...):
    def _print(line):
        lines.append(line)
else:
    def _print(line):
        click.echo(line)

...
_print('{project} - {time}'.format(...)
...

if pager or (...):
    click.echo('\n'.join(lines)
```

Reply:
Done in 046bf4ebcaa8fe44af4927d575f90b6399ef330d